.github/workflows/build.yml

```diff
@@ -84,69 +88,183 @@ jobs:
           restore-keys: |
             ${{ env.CACHE_VERSION }}-cabal-${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles(format('cabal.GHC-{0}.config', matrix.ghc)) }}-
 
-      - shell: bash
+      - id: abc
+        shell: bash
         run: .github/ci.sh build_abc
         if: runner.os != 'Windows'
 
       - shell: bash
         run: .github/ci.sh build
         continue-on-error: false
 
-      - shell: bash
-        run: .github/ci.sh install_system_deps
-        env:
-          Z3_VERSION: "4.8.10"
-          CVC4_VERSION: "4.1.8"
-          YICES_VERSION: "2.6.2"
+      - uses: LisannaAtGalois/.github/actions/cabal-collect-bins@lisanna/cabal-collect-bins
```

Comment:
Placeholder for testing - the action should be merged into the .github repository before this is merged.